drivers/src/mldsa87.rs

```diff
@@ -14,56 +14,85 @@ Abstract:
 #![allow(dead_code)]
 
 use crate::{
-    array::{Array4x1157, Array4x648},
-    Array4x16,
+    array::{Array4x1157, Array4x16, Array4x648, Array4x8},
+    kv_access::{KvAccess, KvAccessErr},
+    wait, CaliptraError, CaliptraResult, KeyReadArgs, Trng,
 };
-use crate::{CaliptraResult, KeyReadArgs, Trng};
+#[cfg(not(feature = "no-cfi"))]
+use caliptra_cfi_derive::cfi_impl_fn;
+use caliptra_registers::mldsa::{MldsaReg, RegisterBlock};
 
 #[must_use]
 #[repr(u32)]
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum MlDsa87Result {
+pub enum Mldsa87Result {
     Success = 0xAAAAAAAA,
     SigVerifyFailed = 0x55555555,
 }
 
 /// MLDSA-87 Public Key
-pub type MlDsa87PubKey = Array4x648;
+pub type Mldsa87PubKey = Array4x648;
 
 /// MLDSA-87 Signature
-pub type MlDsa87Signature = Array4x1157;
+pub type Mldsa87Signature = Array4x1157;
 
 /// MLDSA-87 Message (64 Bytes)
-pub type MlDsa87MsgScalar = Array4x16;
+pub type Mldsa87MsgScalar = Array4x16;
```

Comment:
"Scalar" is not really correct for ML-DSA. In ECDSA, they were the scalar (integer) coordinates of a point, but in ML-DSA, they're just data.

I'd suggest just `Mldsa87Msg` and `Mldsa87SignRnd` or maybe `Mldsa87Data` and `Mldsa87SignRndData`

Reply:
Updated.